packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -351,56 +328,39 @@ export class GutenbergEditorPage {
 		await frame.fill( selectors.blockSearch, text );
 	}
 
-	/**
-	 * @returns Whether the Settings sidebar is open or not.
-	 */
-	async isSettingsSidebarOpen(): Promise< boolean > {
-		const frame = await this.getEditorFrame();
-		return await frame.$eval( selectors.settingsToggle, ( element ) =>
-			element.classList.contains( 'is-pressed' )
-		);
-	}
-
 	/**
 	 * Opens the Settings sidebar.
 	 */
 	async openSettings(): Promise< void > {
-		const frame = await this.getEditorFrame();
-
-		const isSidebarOpen = await this.isSettingsSidebarOpen();
-		if ( ! isSidebarOpen ) {
-			await frame.click( selectors.settingsToggle );
-		}
-		const settingsToggle = await frame.waitForSelector( selectors.settingsToggle );
-		await frame.waitForFunction(
-			( element ) => element.getAttribute( 'aria-pressed' ) === 'true',
-			settingsToggle
-		);
+		await this.editorToolbarComponent.openSettings();
 	}
 
 	/**
 	 * Closes the Settings sidebar.
 	 */
 	async closeSettings(): Promise< void > {
-		const isSidebarOpen = await this.isSettingsSidebarOpen();
-		if ( ! isSidebarOpen ) {
-			return;
+		// On mobile, the settings panel close button is located on the settings panel itself.
+		if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
+			// {@TODO} Temporary measure, to be unified into framelocator based class
+			// once EditorSettingsSidebarComponent is refactored.
+			const editorSettingsSidebarComponent = new EditorSettingsSidebarComponent(
+				await this.getEditorFrame(),
+				this.page
+			);
+			await editorSettingsSidebarComponent.closeSidebar();
 		}
-		const frame = await this.getEditorFrame();
-		await frame.click( selectors.closeSettingsButton );
+		await this.editorToolbarComponent.closeSettings();
```

Comment:
I believe we're missing an `else` here? We don't need to click on any toolbar buttons in the mobile case, right?

Reply:
The else is not necessary here, since the `EditorToolbarComponent.closeSettings()` checks whether the `Settings` is open or not prior to running the `click` action. In this case, after the mobile viewport handling of `await editorSettingsSidebarComponent.closeSidebar()` the check for whether the setting sidebar is open returns a `false`, so the `closeSettings` method short circuits.